doc/specs/#1620 - Installer selector on multiple matches.md

```diff
@@ -0,0 +1,235 @@
+---
+author: Darshan Rander @SirusCodes
+created on: 2022-10-29
+last updated: 2022-11-11
+issue id: 1620
+---
+
+# Spec Title
+
+For [#1620](https://github.com/microsoft/winget-cli/issues/1620)
+
+## Abstract
+
+Currently, the winget.exe client simply lists all the matches and if the user wants to install the app they need to re-run the command with Id.
+
+The UX can be improved here by giving users an option to select the package instead of asking them to re-run the command with Id.
+
+This can be extended to `upgrade`, `show` and `uninstall` commands.
+
+## Inspiration
+
+Whenever I use Winget to install any application there are high chances that there are name collisions with other names and I get multiple suggestions in the table and it's annoying to copy the ID and rerun the command. I guess this is the case for a lot of others as well.
+
+## Solution Design
+
+If we find multiple packages then the user should get a selector.
+
+> It should not be shown if `--diable-interactivity` is set.
+
+### Settings
+
+The user needs to **opt-in** to this feature.
+
+Winget can use `--prompt` and `--no-prompt` flags to enable and disable it respectively.
+
+The user can change the default setting to always show the prompt by -
+
+```pwsh
+"packageDisambiguation": "prompt"
+```
+
+and disable it by -
+
+```pwsh
+"packageDisambiguation": "no-prompt"
+```
+
+### Valid inputs
+
+- Number - It should match with a _row number_ (0 will exit the selector)
+- Text - It should match either the package _Name_ or _Id_
+- `--source` flag - It refines the search with a specific package source. It can only be used if a user enters text or standalone.
+
+### Invalid inputs
+
+If a user enters an invalid value it should show an error asking the user for input again,
+
+Example: `1ab`
+
+```pwsh
+Invalid input, please enter row number, or package name or id.
+```
+
+If a user enters a number that is out of the shown list then they get an error to enter numbers that are in the list.
+
+Example: `23` in the list of `20`` entries
+
+```pwsh
+Please enter row number which are present in entries above.
+```
+
+### Number of results
+
+We should add a `--list-count` flag to customize the number of packages listed in the prompt. By default, the user will see a list of 20 packages.
+
+If the number of packages is more than allowed it should show a text saying how many more packages are available
```

Comment:
How would users see the additional packages if they don't want any of the first 20?

Reply:
I thought about it. I felt it was not necessary but more I think about it I feel it's good to have.

`-m` and `--more` flag is ok I feel.

Will add it in the next iteration.